apps/web/src/services/dashboards/components/dashboard-folder/DashboardFolderDeleteModal.vue

```diff
@@ -0,0 +1,216 @@
+<script lang="ts" setup>
+import { computed, reactive, watch } from 'vue';
+
+import { SpaceConnector } from '@cloudforet/core-lib/space-connector';
+import { PDataTable, PI } from '@cloudforet/mirinae';
+import type { TreeNode } from '@cloudforet/mirinae/src/data-display/tree/tree-view/type';
+
+import type { PrivateDashboardDeleteParameters } from '@/schema/dashboard/private-dashboard/api-verbs/delete';
+import type { PrivateFolderDeleteParameters } from '@/schema/dashboard/private-folder/api-verbs/delete';
+import type { PublicDashboardDeleteParameters } from '@/schema/dashboard/public-dashboard/api-verbs/delete';
+import type { PublicFolderDeleteParameters } from '@/schema/dashboard/public-folder/api-verbs/delete';
+import { i18n } from '@/translations';
+
+import { useDashboardStore } from '@/store/dashboard/dashboard-store';
+
+import { showSuccessMessage } from '@/lib/helper/notice-alert-helper';
+
+import DeleteModal from '@/common/components/modals/DeleteModal.vue';
+import ErrorHandler from '@/common/composables/error/errorHandler';
+import { useProxyValue } from '@/common/composables/proxy-state';
+
+import { gray } from '@/styles/colors';
+
+import { useDashboardMainPageStore } from '@/services/dashboards/stores/dashboard-main-page-store';
+import type { DashboardTreeDataType } from '@/services/dashboards/types/dashboard-folder-type';
+
+
+
+const DELETE_TABLE_FIELDS = [
+    { name: 'name', label: 'Name' },
+    { name: 'location', label: 'Location' },
+];
+type FolderDeleteParams = PublicFolderDeleteParameters | PrivateFolderDeleteParameters;
```

Comment:
nitpick: What about to add file under schema/dashboard which includes those kind of reusable union types?

Reply:
okay!!